src/encoder.rs

```diff
@@ -94,6 +119,30 @@ impl Encoder {
         }
         .check_result()
     }
+
+    /// Get the next page from the stream
+    ///
+    /// Only use if creating encoder with [`Encoder::create_pull`].
+    pub fn get_page(&mut self, flush: bool) -> Option<Vec<u8>> {
+        let mut page_ptr: *mut std::ffi::c_uchar = std::ptr::null_mut();
+        let mut page_size = 0;
+        unsafe {
+            let available = crate::ffi::ope_encoder_get_page(
+                self.ptr,
+                &mut page_ptr,
+                &mut page_size,
+                flush as i32,
+            );
+            if available == 1 {
+                let len = page_size as usize;
+                let page_slice = std::slice::from_raw_parts(page_ptr, len);
+                let page_vec = page_slice.to_vec();
+                Some(page_vec)
```

Comment:
This can be collapsed then.

```suggestion
                Some(std::slice::from_raw_parts(page_ptr, page_size as usize))
```